ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/question_answering/SentenceHighlightingQATranslator.java

```diff
@@ -107,89 +102,188 @@ public NDList processInput(TranslatorContext ctx, Input input) {
+            ctx.setAttachment(MLInput.CONTEXT_FIELD, context);
+
+            // Step 1: Split context into sentences (using full context)
             List<Sentence> sentences = segmenter.segment(context);
             ctx.setAttachment(KEY_SENTENCES, sentences);
-            ctx.setAttachment(MLInput.QUESTION_FIELD, question);
 
-            Encoding encodings = tokenizer.encode(question, context);
+            // Step 2: Create word-level sentence IDs from full context
+            int[] wordLevelSentenceIds = createWordLevelSentenceIds(sentences, context);
 
-            NDArray indicesArray = manager.create(encodings.getIds());
-            indicesArray.setName(INPUT_IDS);
+            // Step 3: Get the target chunk's encoding
+            Encoding targetEncoding = getChunkEncoding(question, context, chunkNumber);
 
-            NDArray attentionMaskArray = manager.create(encodings.getAttentionMask());
-            if (attentionMaskArray.isEmpty()) {
-                throw new IllegalArgumentException("Attention mask is empty in sentence highlighting QA model input");
-            }
-            attentionMaskArray.setName(ATTENTION_MASK);
+            // Step 4: Create sentence IDs array for this chunk
+            int[] sentenceIdsArray = createSentenceIdsArray(targetEncoding, wordLevelSentenceIds, chunkNumber);
 
-            NDArray tokenTypeIdsArray = manager.create(encodings.getTypeIds());
-            tokenTypeIdsArray.setName(TOKEN_TYPE_IDS);
+            // Step 5: Create NDArrays for model input
+            return createModelInputs(manager, targetEncoding, sentenceIdsArray);
 
-            return new NDList(indicesArray, attentionMaskArray, tokenTypeIdsArray);
         } catch (Exception e) {
+            log.error("Error processing input", e);
             throw new IllegalArgumentException(String.format(Locale.ROOT, "Error processing input: %s", e.getMessage()), e);
         }
     }
 
+    /**
+     * Get the encoding for a specific chunk
+     */
+    private Encoding getChunkEncoding(String question, String context, int chunkNumber) {
+        Encoding fullEncoding = tokenizer.encode(question, context);
+
+        if (chunkNumber == 0) {
+            return fullEncoding;
+        } else {
+            Encoding[] overflowEncodings = fullEncoding.getOverflowing();
+            if (overflowEncodings != null && chunkNumber <= overflowEncodings.length) {
+                return overflowEncodings[chunkNumber - 1];
+            } else {
+                throw new IllegalArgumentException("Invalid chunk number: " + chunkNumber);
+            }
+        }
+    }
+
+    /**
+     * Create sentence IDs array for the given chunk
+     */
+    private int[] createSentenceIdsArray(Encoding encoding, int[] wordLevelSentenceIds, int chunkNumber) {
+        long[] wordIds = encoding.getWordIds();
+        int[] sentenceIdsArray = new int[wordIds.length];
+        Arrays.fill(sentenceIdsArray, -100); // Initialize with -100 (ignore token)
```

Comment:
also be nice to extract this default initialization number -100

Reply:
ack